### PR TITLE
Added support for MsSQL2012 ( allow offset )

### PR DESCRIPTION
**Pull request by bckp**
Config now have new option, version that set driver version ( this is not ideal solution, but I do not have better yet ). This added string to the driver name, so switch by driver work as expected and recognize driver with version. Mssql2012 and Sqlsrv2012 now can use offset.

### CODE REVIEW

File: dibi/drivers/DibiPdoDriver.php [hunk 1]
```diff
@@ -396,6 +403,14 @@ public function applyLimit(& $sql, $limit, $offset)
 				}
 				break;
 
+			case 'mssql2012':
+			case 'sqlsrv2012':
+				if($offset >= 0 || $limit >= 0) {
+					$sql .= ' OFFSET ' . (int) $offset . ' ROWS'
```
Comment (dg) [inline dibi/drivers/DibiPdoDriver.php]: missing space after if

File: dibi/drivers/DibiPdoDriver.php [hunk 2]
```diff
@@ -396,6 +403,14 @@ public function applyLimit(& $sql, $limit, $offset)
...
+					. ( $limit > 0 ? ' FETCH NEXT ' . (int) $limit . ' ROWS ONLY' : '' );
```
Comment (dg) [inline dibi/drivers/DibiPdoDriver.php]: extra space after and before ( )

File: dibi/drivers/DibiPdoDriver.php [hunk 3]
```diff
@@ -75,6 +75,9 @@ public function connect(array & $config)
 		}
 
 		$this->driverName = $this->connection->getAttribute(PDO::ATTR_DRIVER_NAME);
+		
+		if (isset($config['version']))
+			$this->driverName .= $config['version'];
 	}
```
Comment (dg) [inline dibi/drivers/DibiPdoDriver.php]: missing }

**Comment by milo** [comment]
From where the `mssql2012` and `sqlsrv2012` names came from? I didn't find any mention in [doc](http://php.net/manual/en/pdo.drivers.php).

**Comment by dg** [comment]
Can you add a test?

**Comment by dg** [comment]
@milo this is probably an extension added by @bckp

**Comment by milo** [comment]
Oh I see :bulb:

**Comment by milo** [comment]
@bckp Isn't here some functionality to fetch database version by some query or PDO feature?

**Comment by bckp** [comment]
@milo good point, I will search for that...

**Comment by bckp** [comment]
@dg I can try add tests

**Comment by bckp** [comment]
@milo Somethink like that??
@dg What is yout opinion for last version?

**Comment by dg** [comment]
Yes, it is better. (Except coding style, ie. spaces, you know)

**Comment by bckp** [comment]
@dg Fixed, with little hope ( should I create MsSql2012Driver with this modifications? )

**Comment by dg** [comment]
Oh, you have a non-standard coding style also in comments :-))

File: dibi/drivers/DibiPdoDriver.php [hunk 4]
```diff
@@ -396,6 +400,17 @@ public function applyLimit(& $sql, $limit, $offset)
...
+						$sql .= ' OFFSET ' . (int) $offset . ' ROWS'
+							. ($limit > 0 ? ' FETCH NEXT ' . (int) $limit . ' ROWS ONLY' : '');
+					}
+					break;
+				}
+				// intentionally break omitted
+
 			case 'odbc':
 			case 'dblib':
 			case 'mssql':
```
Comment (dg) [inline dibi/drivers/DibiPdoDriver.php]: this lines can be removed

Comment (bckp) [inline dibi/drivers/DibiPdoDriver.php]: what lines?

Comment (dg) [inline dibi/drivers/DibiPdoDriver.php]: case 'mssql' and case 'sqlsrv'

**Comment by bckp** [comment]
@dg In comment? Could you point me to the specific location?

**Comment by dg** [comment]
http://www.liteera.cz/slovnik/kulate-zavorky

**Comment by dg** [comment]
Now it looks good, if you can create a test, it would be great.

**Comment by bckp** [comment]
@dg Ja to pochopil že v komentáři v kódu :)

**Comment by bckp** [comment]
@dg I can write tests, but should be better to wrote full PDO mssql test, so maybe later in new PR?

**Comment by dg** [comment]
Later sounds like never ;-)  So what about one test for this one task?

**Comment by bckp** [comment]
@dg like this?

**Comment by dg** [comment]
Great!!

**Comment by bckp** [comment]
@dg Squash it too?

**Comment by dg** [comment]
Yes. One for dump and second for offset with test.

**Comment by bckp** [comment]
@dg ping

**Comment by dg** [comment]
Thx!

**Pull request merged**